R/get_types_recipes.R

```diff
@@ -0,0 +1,112 @@
+## Buckets variables into discrete, mutally exclusive types
+get_types <- function(x) {
+  res <- lapply(x, get_types_recipes)
+  tibble(variable = names(res), type = unname(res))
+}
+
+#' Get types for use in recipes
+#'
+#' The `get_types_recipes()` generic is used internally to supply types to
+#' columns used in recipes. These functions underlie the work that the user sees
+#' in [selections].
+#'
+#' This function acts a extended recipes-specific version of [class()]. By
+#' ignoring differences in similar types ("double" and "numeric") and allowing
+#' each element to have multiple types ("factor" returns "factor", "unordered",
+#' and "nominal". And "character" returns "string", "unordered", and
+#' "nominal") we are able to create more natural selectors such as
+#' [all_nominal()], [all_string()] and [all_integer()].
+#'
+#' @export
+#' @param x A character vector
+#' @examplesIf rlang::is_installed("modeldata")
+#'
+#' data(Sacramento, package = "modeldata")
+#' lapply(Sacramento, get_types_recipes)
+get_types_recipes <- function(x) {
+  UseMethod("get_types_recipes")
+}
+
+#' @export
+#' @rdname get_types_recipes
+get_types_recipes.default <- function(x) {
```

Comment:
Wondering if this would be more expressive / result in less exports as tibble-based lookup. Something like [parsnip's printing](https://github.com/tidymodels/parsnip/blob/ed346749c645b6533b1c94863aa4a1f0fcf7713e/R/print.R#L29-L47).

Reply:
Looking at the deleted code later on, looks like S3 dispatch may make falling back to subclasses a bit easier? On board, if you feel the same way.

Reply:
We need it as a generic since this way we allow other packages to create new methods for types not currently supported, without the need for our involvement :)

Reply:
Got it, thanks!